api/v2/checluster_types.go

```diff
@@ -90,6 +90,16 @@ type CheClusterDevEnvironments struct {
 	// These default components are meant to be used when a Devfile, that does not contain any components.
 	// +optional
 	DefaultComponents []devfile.Component `json:"defaultComponents,omitempty"`
+	// Idle timeout for workspaces, for example: 15m, 30m, 1h.
+	// This timeout is the duration after which a workspace will be idled if there is no activity.
+	// If this value is not provided, workspaces will not idle.
+	// +optional
+	SecondsOfInactivityBeforeIdling string `json:"secondsOfInactivityBeforeIdling,omitempty"`
```

Comment:
Do we have any default value? if so, pls add the following annotation `// +kubebuilder:default:=`
I am thinking about some sort of validation `// +kubebuilder:validation:Pattern=^\d+[ywdhms]$`
WDYT?

Reply:
Hi @l0rd , should the default value for workspace idling be set to `900` (= 15 minutes) ? Then, if the admin wants to remove idling the admin would need to set it to maybe `-1`, how does that sound?

Currently, this PR does has no workspace idling by default

Reply:
Good point @tolusha, we should have default values and they should be set to 1800 (30 mins) and -1 respectively

Reply:
Sounds great, fixed https://github.com/eclipse-che/che-operator/pull/1443#issuecomment-1182466989